### ISSUE TITLE
Multiple Dashboards (Tabs, pages, etc.)

### ISSUE THREAD

**Issue by joepavitt** [FlowFuse/node-red-dashboard#23]

**Comment by joepavitt** [comment]
With the complexities of Express Server routing being very difficult to update, this _may_ need to park for a little while whilst we focus attention on others. 

Dashboard 1.0 doesn't offer this functionality anyway, so this would be improvement on D1.0, not playing catch up.

**Comment by AndreKR** [comment]
I think this is a rather important feature. I've been using Node-RED for several years now but I have never made a dashboard, because since I can only have one I'm saving it for a special occasion.

**Comment by joepavitt** [comment]
> but I have never made a dashboard, because since I can only have one

For clarification you can have multiple, independent, pages in a single "Dashboard". It's just that you can then [navigate between those pages from the navigation menu](https://dashboard.flowfuse.com/nodes/config/ui-base.html#navigation-style-options) - although this can also be hidden.

**Comment by AndreKR** [comment]
Yeah, I played around a bit with pages to see if maybe I can quickly hide and show pages using the "ui control" node based on different dashboard URLs to simulate multiple dashboards, but it didn't really lead to anything.

**Comment by joepavitt** [comment]
Challenge we will have when solving this is that `ui-theme` does not have the concept of an owner, they live freely in Node-RED currently. `ui-page` does have a link to `ui-theme`, so we could utilise that relationship to infer which themes belong to which dashboards.

**Comment by phedders** [comment]
I would be very grateful for this to be possible...

**Comment by ramiws** [comment]
There is now "Add new ui-base" in the dropdown menu. But nothing gets added, is this the expected behaviour? 
I really need this feature as to have one dashboard for us internally, and another for the customer.
Is this the issue to track progress on ability to add new ui-base?

**Comment by sixtybi** [comment]
> I would be very grateful for this to be possible...

Me too!
I just made my dashboard available via nginx with Google SSO authentication, but I can't solve the multi user dashboard. I would like to create a separate dashboard to my kids, to control only their own stuff in the house, but it's not working. This multiple ui-base could solve it maybe. I hope it will be working soon!

**Comment by UniversalBiggi** [comment]
Would be great for me too!
I need to build completely different UI's with sidenavs for different applications / userneeds based on same Database / shared functions which operating same Unit.  So multiple instances of NodeRed is an option but would not be smart, maintainable and performant. Also a lot of duplicatiions and Copy+Paste errors do not smile to me.
I love to hear something from here in future, notifications are subscribed ;)

**Comment by colinl** [comment]
I think you could do this using [node-red-contrib-uibuilder](https://flows.nodered.org/node/node-red-contrib-uibuilder).  Ask on the [forum](https://discourse.nodered.org/) if you want to know how.

**Comment by AndreKR** [comment]
I tried uibuilder once and it was... well, a very different experience than the built-in dashboards. It's very flexible, but it does not really lend itself to quick prototyping.

**Comment by UniversalBiggi** [comment]
> I tried uibuilder once and it was... well, a very different experience than the built-in dashboards. It's very flexible, but it does not really lend itself to quick prototyping.

I dived into that uibuilder documentation last year. It is overkill for my needs... The given frame and structure of FlowFuse dashboard, some nodes and some custom html templates does all i need. Furthermore, I would have to migrate a large part of it to uibuilder, so I will probably continue to wait eagerly for this feature.
